src/openzeppelin/governance/timelock/library.cairo

```diff
@@ -0,0 +1,505 @@
+    end
+
+    func get_min_delay{
+        syscall_ptr: felt*, 
+        pedersen_ptr: HashBuiltin*, 
+        range_check_ptr
+    }() -> (min_delay: felt):
+        let (min_delay: felt) = Timelock_min_delay.read()
+
+        return (min_delay)
+    end
+
+    func get_timestamp{
+        syscall_ptr: felt*, 
+        pedersen_ptr: HashBuiltin*, 
+        range_check_ptr
+    }(id: felt) -> (timestamp: felt):
+        let (timestamp: felt) = Timelock_timestamps.read(id)
+
+        return (timestamp=timestamp)
+    end
+
+    func is_operation{
+        syscall_ptr: felt*, 
+        pedersen_ptr: HashBuiltin*, 
+        range_check_ptr
+    }(id: felt) -> (operation: felt):
+        let (timestamp: felt) = get_timestamp(id)
+        let (operation: felt) = is_not_zero(timestamp)
+
+        return (operation=operation)
+    end
+
+    func is_operation_pending{
+        syscall_ptr: felt*, 
+        pedersen_ptr: HashBuiltin*, 
+        range_check_ptr
+    }(id: felt) -> (pending: felt):
+        alloc_locals
+
+        let (timestamp: felt) = get_timestamp(id)
+        let (pending: felt) = is_le(DONE_TIMESTAMP + 1, timestamp)
+
+        return (pending=pending)
+    end
+
+    func is_operation_ready{
+        syscall_ptr: felt*, 
+        pedersen_ptr: HashBuiltin*, 
+        range_check_ptr
+    }(id: felt) -> (ready: felt):
+        alloc_locals
+
+        let (timestamp: felt) = get_timestamp(id)
+        let (block_timestamp: felt) = get_block_timestamp()
+
+        let (pending: felt) = is_operation_pending(id)
+        let (ready: felt) = is_le(timestamp, block_timestamp)
+
+        if pending + ready == 2:
+            return (TRUE)
+        else:
+            return (FALSE)
+        end
+    end
+
+    func is_operation_done{
+        syscall_ptr: felt*, 
+        pedersen_ptr: HashBuiltin*, 
+        range_check_ptr
+    }(id: felt) -> (done: felt):
+        let (timestamp: felt) = get_timestamp(id)
+
+        if timestamp == DONE_TIMESTAMP:
+            return (TRUE)
+        else:
+            return (FALSE)
+        end
+    end
+
+    func update_delay{
+        syscall_ptr: felt*, 
+        pedersen_ptr: HashBuiltin*, 
+        range_check_ptr
+    }(min_delay: felt):
+        let (old_min_delay: felt) = Timelock_min_delay.read()
+        Timelock_min_delay.write(min_delay)
+        MinDelayChange.emit(old_min_delay, min_delay)
+
+        return ()
+    end
+
+    func schedule{
+        syscall_ptr: felt*, 
+        pedersen_ptr: HashBuiltin*, 
+        range_check_ptr
+    }(
+        call_array_len: felt,
+        call_array: TimelockCall*,
+        calldata_len: felt,
+        calldata: felt*,
+        predecessor: felt,
+        salt: felt,
+        delay: felt,
+    ):
+        alloc_locals
+
+        let (id: felt) = hash_operation(call_array_len, call_array, calldata, predecessor, salt)
+
+        let (calls: Call*) = alloc()
+        _from_timelock_calls_to_calls(call_array_len, call_array, calldata, calls)
```

Comment:
this could also be replaced with Account's `from_call_array_to_call`. probably worth moving it into `_schedule` in case we don't merge the functions as suggested below